src/java.base/share/classes/java/util/zip/ZipFile.java

```diff
@@ -1497,6 +1497,9 @@ private void initCEN(int knownTotal) throws IOException {
                     zerror("invalid END header (bad central directory offset)");
                 }
                 // read in the CEN and END
+                if (end.cenlen + ENDHDR >= Integer.MAX_VALUE) {
+                    zerror("invalid END header (too large central directory size)");
```

Comment:
This check looks correct. It might be a bit clearer to say that "central directory size too large" rather than "too large central directory size".

The bug report says that JDK 8 and the native zip handle these zip files, were you able to check that?

Reply:
I will correct the message as you pointed out.

I checked the JDK8 and Linux unzip commands and found no exceptions or errors. Since JDK9, [JDK-8145260](https://bugs.openjdk.java.net/browse/JDK-8145260) has been modified to significantly change the way zip files are handled, resulting in a different result than jdk8.

Reply:
@AlanBateman Could you please review the above comments.

Reply:
> @AlanBateman Could you please review the above comments.

Thanks for changing the message, the update to ZipFile looks good to me although I'm still surprised that the native tools support a CEN larger than 2Gb.

I'm slow to add myself as a Reviewer because I'm concerned about the test. I ran it on a couple of systems and it take several minutes to create the ZIP file. This test may potentially run concurrently with other I/O bound tests and I'm concerned about intermittent timeouts. @LanceAndersen - do you have any opinions on this?

Reply:
> > @AlanBateman Could you please review the above comments.
> 
> Thanks for changing the message, the update to ZipFile looks good to me although I'm still surprised that the native tools support a CEN larger than 2Gb.
> 
> I'm slow to add myself as a Reviewer because I'm concerned about the test. I ran it on a couple of systems and it take several minutes to create the ZIP file. This test may potentially run concurrently with other I/O bound tests and I'm concerned about intermittent timeouts. @LanceAndersen - do you have any opinions on this?

I haven't  had a chance to look at this yet but will kick off some runs to see what the runtime via mach5.  We might have issues with some of the test machines.

> > @AlanBateman Could you please review the above comments.
> 
> Thanks for changing the message, the update to ZipFile looks good to me although I'm still surprised that the native tools support a CEN larger than 2Gb.
> 
> I'm slow to add myself as a Reviewer because I'm concerned about the test. I ran it on a couple of systems and it take several minutes to create the ZIP file. This test may potentially run concurrently with other I/O bound tests and I'm concerned about intermittent timeouts. @LanceAndersen - do you have any opinions on this?

I have not had a chance to look at this yet in detail but in the meantime I will kick off a mach5 run to see what the runtime is on our test machines today

A couple quick passing comments.    The test should delete the file after the run/error given the potential size.  Was there a reason you did not the test is not TestNG based?